examples_tests/50.MeshPacking/main.cpp

```diff
@@ -0,0 +1,388 @@
+
+    output.vtxBuffer = { 0ull, packerDataStore.vertexBuffer };
+    output.idxBuffer = packerDataStore.indexBuffer;
+    output.indirectDrawBuff = packerDataStore.MDIDataBuffer;
+
+    //create offset tables
+}
+
+void setPipeline(IVideoDriver* driver, IAssetManager* am, SVertexInputParams& vtxInputParams,
+    core::smart_refctd_ptr<IGPURenderpassIndependentPipeline>& gpuPipeline)
+{
+    IAssetLoader::SAssetLoadParams lp;
+
+    auto vertexShaderBundle = am->getAsset("../shader.vert", lp);
+    auto fragShaderBundle = am->getAsset("../shader.frag", lp);
+    ICPUSpecializedShader* shaders[2] =
+    {
+        IAsset::castDown<ICPUSpecializedShader>(vertexShaderBundle.getContents().begin()->get()),
+        IAsset::castDown<ICPUSpecializedShader>(fragShaderBundle.getContents().begin()->get())
+    };
+
+    {
+        auto gpuShaders = driver->getGPUObjectsFromAssets(shaders, shaders + 2);
+        IGPUSpecializedShader* shaders[2] = { gpuShaders->operator[](0).get(), gpuShaders->operator[](1).get() };
+
+        asset::SPushConstantRange pcRange = { asset::ISpecializedShader::ESS_VERTEX, 0u, sizeof(core::matrix4SIMD) };
+        auto pipelineLayout = driver->createGPUPipelineLayout(&pcRange, &pcRange + 1);
+
+        gpuPipeline = driver->createGPURenderpassIndependentPipeline(
+            nullptr, core::smart_refctd_ptr(pipelineLayout),
+            shaders, shaders + 2u,
+            vtxInputParams,
+            asset::SBlendParams(), asset::SPrimitiveAssemblyParams(), SRasterizationParams());
+    }
+}
+
+void setPipelineV2(IVideoDriver* driver, ICPUSpecializedShader* vs, ICPUSpecializedShader* fs,
+    core::smart_refctd_ptr<IGPUBuffer>& vtxBuffer, std::array<SBufferRange<IGPUBuffer>, 2>& dataOffsetBuffers,
+    core::smart_refctd_ptr<IGPUDescriptorSet>& outputGPUDescriptorSet,
+    core::smart_refctd_ptr<IGPURenderpassIndependentPipeline>& outputGpuPipeline)
+{
+    ICPUSpecializedShader* cpuShaders[2] = { vs, fs };
+    auto gpuShaders = driver->getGPUObjectsFromAssets(cpuShaders, cpuShaders + 2);
+
+    core::smart_refctd_ptr<IGPUDescriptorSetLayout> dsLayout;
+    {
+        IGPUDescriptorSetLayout::SBinding b[5];
+        b[0].binding = 0u; b[1].binding = 1u; b[2].binding = 2u; b[3].binding = 3u; b[4].binding = 4u;
+        b[0].type = b[1].type = b[2].type = EDT_UNIFORM_TEXEL_BUFFER;
+        b[3].type = b[4].type = EDT_STORAGE_BUFFER;
+        b[0].stageFlags = b[1].stageFlags = b[2].stageFlags = b[3].stageFlags = b[4].stageFlags = ISpecializedShader::ESS_VERTEX;
+        b[0].count = b[1].count = b[2].count = b[3].count = b[4].count = 1u;
+        dsLayout = driver->createGPUDescriptorSetLayout(b, b + 5u);
+    }
+    
+    asset::SPushConstantRange pcRange = { asset::ISpecializedShader::ESS_VERTEX, 0u, sizeof(asset::SBasicViewParameters) };
+    auto pipelineLayout = driver->createGPUPipelineLayout(&pcRange, &pcRange + 1, core::smart_refctd_ptr(dsLayout));
```

Comment:
pipeline needs all the descriptor layouts (including the ones that come from the loader for textures, etc.)